secretupdater/secretupdater/models.py

```diff
@@ -260,156 +260,176 @@ def _get_k8s_auth(credentials):
+        app.logger.debug(eks_output)
+    except TypeError:
+        raise ClusterAttributeError("Cluster credential or metadata misconfigured or not found")
+
+    try:
+        json_token = json.loads(eks_output)
+        token_data = dict(filter(lambda x: "status" in x, json_token.items()))
+        token = token_data['status']['token']
+        app.logger.debug(token)
+    except ValueError:
+        raise BadEKSToken("Unable to obtain a valid token aws-cli")
+
+    return {
+        "clusters": [
+            {
+                "name": eks_cluster_id,
+                "cluster": {
+                    "server": eks_endpoint,
+                    "certificate-authority-data": _get_credential(kube, 'eks-certificate-authority-data'),
+                },
+            },
+        ],
+        "contexts": [
+            {
+                "name": eks_cluster_id,
+                "context": {
+                    "cluster": eks_cluster_id,
+                    "user": "admin"
+                },
+            }
+        ],
+        "current-context": eks_cluster_id,
+        "users": [
+            {
+                "name": "admin",
+                "user": {
+                    "token": token
+                },
+            },
+        ],
+    }
+
+
+def _get_k8s_auth_token(kube):
+    return {
+        "clusters": [
+            {
+                "name": id,
+                "cluster": {
+                    "server": kube.get('metadata').get('endpoint'),
+                    "insecure-skip-tls-verify": True
+                },
+            },
+        ],
+        "contexts": [
+            {
+                "name": id,
+                "context": {
+                    "cluster": id,
+                    "user": "admin"
+                },
+            }
+        ],
+        "current-context": id,
+        "users": [
+            {
+                "name": "admin",
+                "user": {
+                    "token": _get_credential(kube, 'token')
+                }
+            }
+        ]
+    }
+
+
+def _get_k8s_auth_userpass(kube):
+    return {
+        "clusters": [
+            {
+                "name": id,
+                "cluster": {
+                    "server": kube.get('metadata').get('endpoint'),
+                    "insecure-skip-tls-verify": True,
+                },
+            },
+        ],
+        "contexts": [
+            {
+                "name": id,
+                "context": {
+                    "cluster": id,
+                    "user": "admin"
+                },
+            }
+        ],
+        "current-context": id,
+        "users": [
+            {
+                "name": "admin",
+                "user": {
+                    "username": _get_credential(kube, 'username'),
+                    "password": _get_credential(kube, 'password'),
+                },
+            }
+        ],
+    }
+
```

Comment:
Why you do this to me :)

Reply:
because Python code style requires to blank lines ?

Reply:
Guess I learned something today. Another reason I couldn't use Python :)